Cache decoded toga color on gtk (#3117)
beeware/toga: changes/3117.misc.rst

> On gtk, the decoded toga color is now cached for faster lookup.

Reviewer: ```suggestion
On GTK, native colors representations are now cached.
```
